spark/src/main/scala/org/apache/comet/serde/QueryPlanSerde.scala

```diff
@@ -202,7 +202,9 @@ object QueryPlanSerde extends Logging with ShimQueryPlanSerde {
       inputs: Seq[Attribute],
       binding: Boolean): Option[AggExpr] = {
     aggExpr.aggregateFunction match {
-      case s @ Sum(child, _) if sumDataTypeSupported(s.dataType) =>
+      case s @ Sum(child, evalMode)
+          if sumDataTypeSupported(s.dataType) &&
+            evalMode == EvalMode.LEGACY =>
```

Comment:
Hmm, this is introduced in Spark 3.4. So Spark 3.2/3.3 builds are failed due to this change. I need to add a shim layer...